dev-libs/re2/re2-0.2018.02.01.recipe

```diff
@@ -0,0 +1,73 @@
+SUMMARY="A Regular Expression Library"
+DESCRIPTION="RE2 is a fast, safe, thread-friendly alternative to backtracking \
+regular expression engines like those used in PCRE, PERL and Python. It is a \
+C++ library."
+HOMEPAGE="http://github.com/google/re2"
+COPYRIGHT="2009 The RE2 Authors"
+LICENSE="BSD (3-clause)"
+REVISION="1"
+RE_VERSION="${portVersion//./-}"
+RE_VERSION="${RE_VERSION#0-}"
+SOURCE_URI="https://github.com/google/re2/archive/$RE_VERSION.tar.gz"
+CHECKSUM_SHA256="c8ab833081c9766ef4e4d1e6397044ff3b20e42be109084b50d49c161f876184"
+SOURCE_FILENAME="re2-$portVersion.tar.gz"
+SOURCE_DIR="re2-$RE_VERSION"
+PATCHES="re2-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	re2${secondaryArchSuffix} = $portVersion
+	lib:libre2${secondaryArchSuffix} = 0.0.0 compat >= 0
```

Comment:
This might be hand
```diff
 ARCHITECTURES="!x86_gcc2 x86 x86_64"
 SECONDARY_ARCHITECTURES="x86"
+
+libVersion="0.0.0"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
PROVIDES="
-	re2${secondaryArchSuffix} = $portVersion
+	re2$secondaryArchSuffix = $portVersion
-	lib:libre2${secondaryArchSuffix} = 0.0.0 compat >= 0
+	lib:libre2$secondaryArchSuffix = $libVersionCompat
 	"
```